hazelcast/src/main/java/com/hazelcast/query/impl/BitmapIndexStore.java

```diff
@@ -403,6 +416,64 @@ Comparable canonicalizeScalarForStorage(Comparable value) {
         return value;
     }
 
+    /**
+     * Accounts insert operations
+     *
+     * @param operationStats current index stats
+     * @param values         iterator which points on first value should be inserted
+     */
+    private void accountEntryInsert(IndexOperationStats operationStats, Iterator values) {
```

Comment:
Can we rewrite this method like this? So there will be no case to create meta object like iterator. 
```suggestion
private void accountEntryInsert(IndexOperationStats operationStats, Object value) {
        if (value instanceof MultiResult) {
            List results = ((MultiResult) value).getResults();
            for (int i = 0; i < results.size(); i++) {
                operationStats.onEntryAdded(null, results.get(i));
            }
        } else {
            operationStats.onEntryAdded(null, value);
        }
    }
```